### PR TITLE
refactor(meta): throttle hummock vacuum

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR throttles vacuum job (both for stale metadata and object), to avoid vacuum job monopolizing meta node's resource. Previously we've encountered a case where heavy vacuum job caused etcd client keep alive failure.

This PR also splits the vacuum job into two separate tokio tasks for metadata and object, so that they won't block each other.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/hummock/vacuum.rs [hunk 1]
```diff
@@ -112,6 +116,10 @@ where
         let batch_size = 500usize;
         let mut sent_batch = Vec::with_capacity(objects_to_delete.len());
         while batch_idx < objects_to_delete.len() {
+            if batch_idx != 0 {
```
Comment (Li0k) [inline src/meta/src/hummock/vacuum.rs:119]: minor: check self.env.opts.vacuum_spin_interval_ms might avoid a useless await (e.g. sleep(0))

File: src/meta/src/hummock/mod.rs [hunk 2]
```diff
@@ -49,20 +48,26 @@ pub fn start_hummock_workers<S>(
...
-    let mut workers = vec![start_checkpoint_loop(
-        hummock_manager,
-        Duration::from_secs(meta_opts.hummock_version_checkpoint_interval_sec),
-        meta_opts.min_delta_log_num_for_hummock_version_checkpoint,
-    )];
-    workers.push(start_vacuum_scheduler(
-        vacuum_manager,
-        Duration::from_secs(meta_opts.vacuum_interval_sec),
-    ));
+    let workers = vec![
```
Comment (Li0k) [inline src/meta/src/hummock/mod.rs:51]: add comments

File: src/common/src/config.rs [hunk 3]
```diff
@@ -741,6 +747,10 @@ pub mod default {
             30
         }
 
+        pub fn vacuum_spin_interval_ms() -> u64 {
```
Comment (Li0k) [inline src/common/src/config.rs:750]: Is the default value of vacuum_spin_interval_ms too large?  Is there a reference for this value?

**Review by Li0k — APPROVED** [review]
Rest, LGTM

File: src/common/src/config.rs (hunk 3 above)
Comment (zwang28) [reply src/common/src/config.rs:750]: Reduced to 10ms.

File: src/meta/src/hummock/mod.rs (hunk 2 above)
Comment (zwang28) [reply src/meta/src/hummock/mod.rs:51]: Comments added.

**Pull request merged**